Improved SWTBot test cases

PhilMFischer opened the pull request:
- Introduced new Condition to wait for Workspace Builders
- Improved Code on calling the inheritance
- Removed unrealistic save operations, also disturbing the call to the
inheritance builder
- Fixed test cases such as calculation along the changes
- Also fixed an NPE in the saveables provider
- Also fixed incorrectly named units in QUDV test cases
---
Closes #283: Further SWTBot improvements

Commit:
Cleaned Up a test case - (Task #283)

- Changed Unit labels were not set correctly

---
Task #283: Further SWTBot improvements

Commit:
Fixed NPE on Savables Provider - (Task #283)

- Saw a case on SWTBot where the saveables provider got called with an
empty list of resources. Accessing it was creating a NPE.

---
Task #283: Further SWTBot improvements

Commit:
Improving SWTBOT test Cases - (Task #283)

- Introduced new Condition to wait for Workspace Builders
- Improved Code on calling the inheritance
- Removed unrealistic save operations, also disturbing the call to the
inheritance builder
- Fixed test cases such as calculation along the changes

---
Task #283: Further SWTBot improvements

Commit:
Improved Code - (Task #283)

- improved readability
- improved reliability

---
Task #283: Further SWTBot improvements

Commit:
Merge branch 'development' of
https://github.com/virtualsatellite/VirtualSatellite4-Core.git into
feature/task_283-Further_SWTBot_improv

Commit:
Remove spotbugs warning - (Task #283)

- Thread should not go to sleep when holding a lock. Instead the thread
is now send to sleep freeing the lock in the given moment.

---
Task #283: Further SWTBot improvements

Commit:
Merge remote-tracking branch 'origin/development' into feature/task_283-Further_SWTBot_improv

Commit:
Got SWTBot tests running on windows - (Task #283)

- New ProjectWaitUnit needs a save action before, so that the builders
are actually triggered.

---
Task #283: Further SWTBot improvements

Commit:
Applied minor fixes to catch NPE with ZMQ and VTK - (Task #318)

- Fixed call from plugin Activator to ZMQ to be cateched in case of link
exception.
- Fixed call from plugin activation to VTK in ClientUpdatedHandler in
case of link exception.

---
Task #318: NPE when executing under linux with incorrect VTK and ZMQ

Commit:
Some Clean Up - (Task #318)

- Added some further logging to caught exception.
- Fixed some white spacing.
---
Task #318: NPE when executing under linux with incorrect VTK and ZMQ

Commit:
Fixed some white space - (Task #283)

---
Task #283: Further SWTBot improvements

Commit:
Merge remote-tracking branch 'origin/development' into feature/task_283-Further_SWTBot_improv

Commit:
Verify that minimum calls have reached the scheduler - (Task #283)

- Currently checking that the worker/builder get triggered only once.

---
Task #283: Further SWTBot improvements

Commit:
Increased Timeout for sync of UI - (Task #283)

- Reset sync time to 100 ms.

---
Task #283: Further SWTBot improvements

Commit:
Merged with Development - (Task #283)

- Resolved issue in Activator of Visualization UI.

---
Task #283: Further SWTBot improvements

andriikovalov-dlr commented on de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java:
```diff
@@ -501,10 +495,10 @@ public void run() {
 		 */
 		void waitForExecution() {
-			synchronized (gotExecuted) {
+			synchronized (this) {
```
Do we really need synchronization here? Maybe we could just declare gotExecuted volatile and do thread sleep instead of wait and all synchronization stuff?

PhilMFischer replied:
By experience it is generally good to stick with the basic synchronization mechanisms.

andriikovalov-dlr commented on de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/util/VirSatWaitForProjectBuilder.java:
```diff
@@ -0,0 +1,185 @@
+ * This Condition waits for the Workspace Builder to execute
+ *
+ */
+public class VirSatWaitForProjectBuilder extends DefaultCondition implements IJobChangeListener {
```
just wondering, can we maybe use (or extend) `org.eclipse.swtbot.eclipse.finder.waits.WaitForJobs`?

PhilMFischer replied:
We used to use the WaitForJobs. This is an improvement which counts the jobs being called and the jobs being stopped.

Commit:
Merge remote-tracking branch 'origin/development' into feature/task_283-Further_SWTBot_improv

PhilMFischer commented on de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java:
```diff
@@ -485,13 +478,14 @@ public static Repository getRepository(IProject project) {
 	 */
 	static class WaitForRunnable implements Runnable {
+		
 		Boolean gotExecuted = false;
```
right.

PhilMFischer commented on de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java:
```diff
@@ -485,13 +478,14 @@ public static Repository getRepository(IProject project) {
 		@Override
 		public void run() {
-			synchronized (gotExecuted) {
+			synchronized (this) {
```
Sure I will consider this change.

Commit:
Applied fixes due to review - (Task #283)

---
Task #283: Further SWTBot improvements

SaMuellerDLR commented on de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java:
```diff
@@ -336,18 +330,17 @@ protected IProject createProjectWithConcepts(String projectName) {
-		bot.saveAllEditors();
-		waitForEditingDomainAndUiThread();
-		bot.waitUntil(Conditions.waitForJobs(ResourcesPlugin.FAMILY_AUTO_BUILD, "eclipse auto builders (calculation builder)"));
+		propagateInheritance();
```
waitForCalculationBuilder() calling propagateInheritance() looks weird.
Either something is weird in the logic or the names should reflect the logic changes that happened.

PhilMFischer replied:
Right removed method and renamed one stating to wait for all builders and ui thread.

Commit:
Fixed method name due to review - (Task #283)

- Changed name of method stating that the call will wait for all
builders to finish execution.

---
Task #283: Further SWTBot improvements

SaMuellerDLR reviewed (approved):
Approved!
